atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTableTest.java

```diff
@@ -225,25 +218,158 @@ public void dropTablesDoesNotThrowIfMultipleMatchingTableReferencesExistAndCaseS
             throws TableMappingNotFoundException {
         // This mock is never called on the case sensitive code path, but this makes the test easier to read.
         lenient().when(sqlConnection.selectCount(any(), any(), any())).thenReturn((long) 2);
-        createTable();
-        createOverflowTable();
+        createTableAndOverflow();
 
         assertThatCode(() -> tableMappingDdlTable.drop()).doesNotThrowAnyException();
     }
 
+    @Test
+    public void createAltersTableIfConfiguredAndOverflowTableExists() throws TableMappingNotFoundException {
+        createTableAndOverflow();
+        setTableToHaveOverflowColumn(false);
+        setMetadataToHaveOverflow(true);
+        tableMappingDdlTable.create(createMetadata(true));
+        verifyNumberOfTimesTableAltered(1);
+    }
+
+    @Test
+    public void createDoesNothingWhenAlterSpecifiedButOverflowColumnExists() throws TableMappingNotFoundException {
+        createTableAndOverflow();
+        setTableToHaveOverflowColumn(true);
+        setMetadataToHaveOverflow(true);
+        assertThatCode(() -> tableMappingDdlTable.create(createMetadata(true))).doesNotThrowAnyException();
+        verifyNumberOfTimesTableAltered(0);
+    }
+
+    @Test
+    public void createDoesNothingWhenAlterSpecifiedButMigrationUnstarted() throws TableMappingNotFoundException {
+        createTableAndOverflow();
+        setTableToHaveOverflowColumn(false);
+        setMetadataToHaveOverflow(false);
+        OracleDdlTable ddlTable = createOracleDdlTable(ImmutableOracleDdlConfig.builder()
+                .from(TABLE_MAPPING_DEFAULT_CONFIG)
+                .overflowMigrationState(OverflowMigrationState.UNSTARTED)
+                .build());
+        assertThatThrownBy(() -> ddlTable.create(createMetadata(true)))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasMessageContaining(MISSING_OVERFLOW_EXCEPTION_MESSAGE);
+        verifyNumberOfTimesTableAltered(0);
+    }
+
+    @Test
+    public void createDoesNothingWhenAlterSpecifiedButOverflowColumnIsNotNeeded() throws TableMappingNotFoundException {
+        createTableAndOverflow();
+        setTableToHaveOverflowColumn(false);
+        setMetadataToHaveOverflow(false);
+        assertThatCode(() -> tableMappingDdlTable.create(createMetadata(false))).doesNotThrowAnyException();
+        verifyNumberOfTimesTableAltered(0);
+    }
+
+    @Test
+    public void createDoesNothingIfOverflowColumnNeededButTableNotListedInConfig()
+            throws TableMappingNotFoundException {
+        createTableAndOverflow();
+        setTableToHaveOverflowColumn(false);
+        setMetadataToHaveOverflow(false);
+        OracleDdlTable ddlTable = createOracleDdlTable(ImmutableOracleDdlConfig.builder()
+                .from(TABLE_MAPPING_DEFAULT_CONFIG)
+                .alterTablesOrMetadataToMatch(Set.of())
+                .build());
+        assertThatThrownBy(() -> ddlTable.create(createMetadata(true)))
```

Comment:
nit: `createDoesNothing` implies that we're not expecting an exception here :) Lets clarify what you do want to test, and update the name and test accordingly :)

Reply:
`createDoesNotAlterTableIfOverflowColumnNeededButTableNotListedInConfig` ?